### ISSUE TITLE
[BUG] Rancher is showing k8s versions < 1.23 for kev1 AKS/GKE clusters

### ISSUE THREAD

**Issue by thaneunsoo** [rancher/dashboard#7277]
**Setup**
- Rancher version: 2.7.0-rc6
- Browser type & version: Chrome Build 106

**Describe the bug**

Rancher is showing k8s versions less than 1.23 for kev1 AKS and GKE clusters

**To Reproduce**

AKS
1. Install Rancher 2.7.0-rc6
2. Create kev1 AKS cluster by accessing the cluster from the URL - https://<server>/g/clusters/add/launch/azureaks
3. View k8s dropdown on UI

GKE
1. Install Rancher 2.7.0-rc6
2. Create kev1 GKE cluster by accessing the cluster from the URL - https://<server>/g/clusters/add/launch/googlegke
3. View k8s dropdown on UI

**Result**
Rancher shows k8s versions less than 1.23 for both AKS and GKE kev1 clusters. The behavior is not reproducible on kev2 clusters

**Expected Result**

k8s versions less than 1.23 should not be shown in the UI.

**Screenshots**

AKS
![image](https://user-images.githubusercontent.com/108360907/197610576-4bc05509-c505-4580-80b3-16ec5b1f9daa.png)

GKE
![image](https://user-images.githubusercontent.com/108360907/197610606-117d3258-487e-4feb-8a3d-a13f76bb603d.png)

**Comment by catherineluse** [comment]
I tried to reproduce this in v2.6-head but couldn't. Here's what I tried:

1. Installed Rancher v2.6-rc7
2. Went to the `/g/clusters/add/launch/azureaksv2` endpoint and installed an AKS cluster. (As discussed, we don't normally support this endpoint, but are using it mainly to simulate a cluster installed in Rancher 2.5.x who should have an upgrade path in the new UI)
3. Confirmed that v1.22.x is not available as a Kubernetes option

<img width="784" alt="Screen Shot 2022-10-26 at 11 18 23 AM" src="https://user-images.githubusercontent.com/20599230/198109847-cc89030f-7914-4896-927a-92643fa0b270.png">
4. Finished provisioning the cluster, went to the dashboard, went to edit the cluster, confirmed the K8s options are the same. (I wouldn't expect any discrepancy between these versions because the same component is reused in both pages)

Then the same for GKE:
1. Went to the `/g/clusters/add/launch/googlegkev2`
2. Confirmed that v1.22.x is not available as a Kubernetes option
<img width="775" alt="Screen Shot 2022-10-26 at 11 39 15 AM" src="https://user-images.githubusercontent.com/20599230/198110559-1a1d1b40-f26e-41f4-9b32-ad82e4783a6f.png">
3. After provisioning the cluster, went to edit the cluster and confirmed that the K8s options are the same.
